### PR TITLE
[robin-map] Fix error when use find_package(robin-map CONFIG REQUIRED) 

### LINKED ISSUE #26597: [robin-map] Rename to tsl-robin-map to match other tessil libraries and fix find_package
**Is your feature request related to a problem? Please describe.**

Yes. The package name exported by the tessil robin-map library is `tsl-robin-map`, and the port name don't match other ports.

Also, the suggested CMake code to add robin map to a project is faulty:

```
robin-map provides CMake targets:
  # this is heuristically generated, and may not be correct
  find_package(robin-map CONFIG REQUIRED)
  target_link_libraries(main PRIVATE tsl::robin_map)
```

But `find_package(robin-map CONFIG REQUIRED)` don't work as expected. This is easily explained by the weird directory structure it creates:

```
vcpkg_installed/arm64-osx/share/
└── robin-map
    ├── copyright
    ├── tsl-robin-mapConfig.cmake
    ├── tsl-robin-mapConfigVersion.cmake
    ├── tsl-robin-mapTargets.cmake
    ├── vcpkg.spdx.json
    └── vcpkg_abi_info.txt
```

As you can see, the `tsl-robin-mapConfig.cmake` is contained in the `robin-map` directory, preventing CMake from finding the file.

**Proposed solution**

Rename `robin-map` to `tsl-robin-map` (possibly with a package that references the old name for backward compat) and fix the directory name when installing tessil robin-map

**Describe alternatives you've considered**

No renaming and adding a symbolic link so CMake can find the config file is also an option, as long as there's a way to find the package.

### PR DESCRIPTION
**Pull request by Cheney-W**
**Describe the pull request**

- #### What does your PR fix?
  Fixes https://github.com/microsoft/vcpkg/issues/26597
  
  Adding `PACKAGE_NAME tsl-robin-map` into the `vcpkg_cmake_config_fixup(CONFIG_PATH share/cmake/tsl-robin-map)` function.
  
  No feature need to test.

### CODE REVIEW

**Comment by JackBoosY** [comment]
No need to rename port, just modify this line:
https://github.com/microsoft/vcpkg/blob/c0b6d35a67d2ad358a6ced92b7aad16a7bf17737/ports/robin-map/portfile.cmake#L14
Add parameter `PACKAGE_NAME tsl-robin-map`.

File: ports/robin-map/portfile.cmake [hunk 1]
```diff
@@ -1,18 +1 @@
...
-    SOURCE_PATH "${SOURCE_PATH}"
-)
-
-vcpkg_cmake_install()
-vcpkg_cmake_config_fixup(CONFIG_PATH share/cmake/tsl-robin-map)
-
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug")
-
-file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
+set(VCPKG_POLICY_EMPTY_PACKAGE enabled)
```
Comment (JackBoosY) [inline ports/robin-map/portfile.cmake:1]: Please add an empty line here.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — DISMISSED** [review]

**Comment by gracicot** [comment]
I just tested it and everything works as expected!

File: ports/tsl-robin-map/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    SHA512 5741049287fdb9c3316e1eb84b99343efc7b35f492e1db8166d65c2d16c7905f51b11cf164bedae9e44d4b6000bbea3c49012acf725a977e665a8dc23e89b1fb
+    HEAD_REF master
+)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+)
+
+vcpkg_cmake_install()
+vcpkg_cmake_config_fixup(CONFIG_PATH share/cmake/tsl-robin-map)
```
Comment (dan-shaw) [inline ports/tsl-robin-map/portfile.cmake:14]: I do not think we should rename the library, instead `vcpkg_cmake_config_fixup` needs to add `PACKAGE_NAME`. Renaming the library will cause versioning issues, so we try not to do that.

Comment (Cheney-W) [reply ports/tsl-robin-map/portfile.cmake]: Done!

**Comment by gracicot** [comment]
Any news on this one?

**Pull request merged**